src/webgpu/api/validation/capability_checks/features/texture_formats.spec.ts

```diff
@@ -89,6 +88,45 @@ g.test('texture_view_descriptor')
     });
   });
 
+g.test('canvas_configuration')
+  .desc(
+    `
+  Test configuring a canvas with optional texture formats will fail if the required optional
+  feature is not enabled.
+
+  Note: This test should be skipped until GPUCanvasConfiguration allow optional formats.
```

Comment:
As mentioned in the TODO comment, this test is being skipped because the canvas configuration doesn't allow any optional formats. So, I'm not sure if we can add the test although the tests are skipped. WDYT?

Reply:
We can test this now. There are no optional formats that are allowed for configure():
https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-configure
BUT note that that validation ("Supported context formats must contain configuration.format.") generates a validation error, not an exception.

So what this test does is, for the optional formats, make sure exceptions are thrown (not validation errors).
Since an exception is thrown prior to the device-timeline validation, there also will be no validation error.

Reply:
Ah, thank you for explaining it! I missed catching it in the specification.